packages/core/src/lib/light_push/light_push_v3.ts

```diff
@@ -0,0 +1,300 @@
+        protoMessage as WakuMessage,
+        encoder.pubsubTopic
+      );
+      return { query, error: null };
+    } catch (error) {
+      log.error("Failed to prepare push message", error);
+
+      return {
+        query: null,
+        error: ProtocolError.GENERIC_FAIL
+      };
+    }
+  }
+
+  public async send(
+    encoder: IEncoder,
+    message: IMessage,
+    peerId: PeerId
+  ): Promise<CoreProtocolResultWithMeta> {
+    const { query, error: preparationError } = await this.preparePushMessage(
+      encoder,
+      message
+    );
+
+    if (preparationError || !query) {
+      return {
+        success: null,
+        failure: {
+          error: preparationError,
+          peerId
+        },
+        protocolUsed: LightPushCodecV3
+      };
+    }
+
+    let stream: Stream;
+    try {
+      stream = await this.getStream(peerId);
+    } catch (error) {
+      log.error("Failed to get stream", error);
+      return {
+        success: null,
+        failure: {
+          error: ProtocolError.NO_STREAM_AVAILABLE,
+          peerId: peerId
+        },
+        protocolUsed: LightPushCodecV3
+      };
+    }
+
+    let res: Uint8ArrayList[] | undefined;
+    try {
+      res = await withTimeout(
+        pipe(
+          [query.encode()],
+          lp.encode,
+          stream,
+          lp.decode,
+          async (source) => await all(source)
+        ),
+        STREAM_TIMEOUT_MS
+      );
+    } catch (err) {
+      log.error("Failed to send waku light push request", err);
+      return {
+        success: null,
+        failure: {
+          error: ProtocolError.STREAM_ABORTED,
+          peerId: peerId
+        },
+        protocolUsed: LightPushCodecV3
+      };
+    } finally {
+      if (stream) {
+        void stream.close();
+      }
+    }
+
+    const bytes = new Uint8ArrayList();
+    res.forEach((chunk) => {
+      bytes.append(chunk);
+    });
+
+    let response: proto_lightpush_v3.LightpushResponse | undefined;
+    try {
+      response = proto_lightpush_v3.LightpushResponse.decode(bytes);
+    } catch (err) {
+      log.error("Failed to decode push response", err);
+      return {
+        success: null,
+        failure: {
+          error: ProtocolError.DECODE_FAILED,
+          peerId: peerId
+        },
+        protocolUsed: LightPushCodecV3
+      };
+    }
+
+    if (!response) {
+      log.error("Remote peer fault: No response received");
+      return {
+        success: null,
+        failure: {
+          error: ProtocolError.NO_RESPONSE,
+          peerId: peerId
+        },
+        protocolUsed: LightPushCodecV3
+      };
+    }
+
+    if (!isValidRequestId(response.requestId)) {
+      log.error("Invalid request ID format", {
+        received: response.requestId
+      });
+      return {
+        success: null,
+        failure: {
+          error: ProtocolError.DECODE_FAILED,
+          peerId: peerId
+        },
+        protocolUsed: LightPushCodecV3,
```

Comment:
let't not change return type and add metadata

Reply:
actually let's remove only `protocolUsed` from Core protocols 
and keep other types of `LightPush` from new spec

that means let's drop `CoreProtocolResultWithMeta` type and introduce new one `LightPushResult`
type from SDK shouldn't change